test/controllers/api_users/permissions_controller_test.rb

```diff
@@ -160,24 +160,46 @@ class ApiUsers::PermissionsControllerTest < ActionController::TestCase
       assert_same_elements [other_permission, not_from_ui_permission], api_user.supported_permissions
     end
 
-    should "prevent permissions being added for other apps" do
-      other_application = create(:application, with_non_delegatable_supported_permissions: %w[other])
-      application = create(:application)
-      api_user = create(:api_user)
-      create(:access_token, application:, resource_owner_id: api_user.id)
+    should "when updating permissions for app A, prevent additionally adding or removing permissions for app B" do
+      application_a = create(:application)
+      application_a_old_permission = create(:supported_permission, application: application_a)
+      application_a_new_permission = create(:supported_permission, application: application_a)
+
+      application_b = create(:application)
+      application_b_old_permission = create(:supported_permission, application: application_b)
+      application_b_new_permission = create(:supported_permission, application: application_b)
+
+      api_user = create(:api_user,
+                        with_signin_permissions_for: [application_a, application_b],
+                        with_permissions: {
+                          application_a => [application_a_old_permission.name],
+                          application_b => [application_b_old_permission.name],
+                        })
+      create(:access_token, application: application_a, resource_owner_id: api_user.id)
+      create(:access_token, application: application_b, resource_owner_id: api_user.id)
```

Comment:
I agree this would be useful, but would mean we can't use the transient params for a few other tests as discussed [here](https://github.com/alphagov/signon/pull/3121#discussion_r1733294274). Leaving this one for now.